util/tlgen/tb__xbar_connect.sv.tpl

```diff
@@ -21,10 +21,10 @@ initial force dut.${rst} = rst_n;
 
 // Host TileLink interface connections
 % for node in xbar.hosts:
-`CONNECT_TL_HOST_IF(${node.name}, dut, ${node.clocks[0]}, rst_n)
+`CONNECT_TL_HOST_IF(${node.name.replace('.', '__')}, dut, ${node.clocks[0]}, rst_n)
```

Comment:
for all the `replace` calls everywhere, does it make sense to have a separate form constructed?

Reply:
Yeah, maybe. The problem is that this is lots of different template files, so they'd all need to include some library etc. I was leaning towards the "simple, stupid" option, but I could happily write a "escape_node_name" function in some library in tlgen. What do you think would be better?

Reply:
ah honestly you're the python expert here, so i trust that you have the best idea.
I was mostly just wondering if we'll ever get into the situation where we need to change something and it becomes a pain to chase down these various bits.  

We could always worry about if the situation ever surfaces though.

Reply:
I know what you mean, but it wasn't all that bad to hunt down the cases to add this, so I'm hopeful that it won't be an issue :-) As you say, let's worry about it on the next rename.

Reply:
Good point, thanks!